static/js/markdown.js

```diff
@@ -129,10 +128,21 @@ export function apply_markdown(message) {
                 */
                 full_name = match[1];
                 user_id = Number.parseInt(match[2], 10);
-
-                if (!helpers.is_valid_full_name_and_user_id(full_name, user_id)) {
-                    user_id = undefined;
-                    full_name = undefined;
+                if (full_name && user_id) {
+                    // For @**user|id** syntax
+                    if (!helpers.is_valid_full_name_and_user_id(full_name, user_id)) {
+                        user_id = undefined;
+                        full_name = undefined;
+                    }
+                } else if (!full_name && user_id) {
```

Comment:
Let's do `full_name === ""` -- we like to avoid using `!foo` in JavaScript because it's ambiguous about whether `undefined`, `null`, `""` are considered, and it's better to be explicit about these things.  

I'd also consider putting this case first, and/or generally structurally it with an outer `if (user_id)` for better readability.

Reply:
I have updated this pr in following manner:
* Putting both these cases inside a outer `if (user_id)` block.
* Checking for `**|user_id**` syntax before **full_name|user_id**.
Please have a look at it.